diff --git a/RL2-Source/GameManager.cs b/RL2-Source-patched/GameManager.cs
index f265753..569b163 100755
--- a/RL2-Source/GameManager.cs
+++ b/RL2-Source-patched/GameManager.cs
@@ -2,6 +2,7 @@ using System.Collections;
 using RL_Windows;
 using Steamworks;
 using UnityEngine;
+using RL2.ModLoader;
 
 public class GameManager : MonoBehaviour
 {
@@ -82,6 +82,8 @@ public class GameManager : MonoBehaviour
 
 	private void OnApplicationQuit()
 	{
+		ModLoader.Log("-----|| Unloading mods ||-----");
+		Messenger<ModMessenger, ModLoaderEvent>.Broadcast(ModLoaderEvent.Unload, null, null);
 		Debug.Log("Application is closing...");
 		IsApplicationClosing = true;
 		Rewired_RL.SetGamepadColor(Color.clear);
diff --git a/RL2-Source/OnGameLoadManager.cs b/RL2-Source-patched/OnGameLoadManager.cs
index b7eaa45..6e7601b 100755
--- a/RL2-Source/OnGameLoadManager.cs
+++ b/RL2-Source-patched/OnGameLoadManager.cs
@@ -2,6 +2,7 @@ using System.Globalization;
 using System.IO;
 using SceneManagement_RL;
 using UnityEngine;
+using RL2.ModLoader;
 
 public class OnGameLoadManager
 {
@@ -39,6 +40,12 @@ public class OnGameLoadManager
 		Application.focusChanged -= OnFocusChanged;
 		Application.focusChanged += OnFocusChanged;
 		ConfineMouseToGameWindow(!SaveManager.ConfigData.DisableCursorConfine);
+		ModLoader.LoadMods();
+		foreach (var mod in ModLoader.LoadedMods)
+		{
+			mod.Load();
+		}
+		Messenger<ModMessenger, ModLoaderEvent>.Broadcast(ModLoaderEvent.Load, null, null);
 	}
 
 	private static LanguageType GetSystemLanguageType()
