diff --git a/RL2-Source/GameManager.cs b/RL2-Source-patched/GameManager.cs
index 569b163..d852696 100755
--- a/RL2-Source/GameManager.cs
+++ b/RL2-Source-patched/GameManager.cs
@@ -48,6 +48,7 @@ public class GameManager : MonoBehaviour
 		{
 			Instance = this;
 			m_pauseEventArgs = new GamePauseStateChangeEventArgs(isPaused: false);
+			gameObject.AddComponent<UnityHook>();
 		}
 		else if (Instance != this)
 		{
