diff --git a/RL2-Source/DoorLocation.cs b/RL2-Source-patched/DoorLocation.cs
index 519c744..e4120f7 100755
--- a/RL2-Source/DoorLocation.cs
+++ b/RL2-Source-patched/DoorLocation.cs
@@ -81,6 +81,9 @@ public struct DoorLocation : IEquatable<DoorLocation>
 
 	public override int GetHashCode()
 	{
-		return (-785574603 * -1521134295 + RoomSide.GetHashCode()) * -1521134295 + DoorNumber.GetHashCode();
+		unchecked
+		{
+			return (-785574603 * -1521134295 + RoomSide.GetHashCode()) * -1521134295 + DoorNumber.GetHashCode();
+		}
 	}
 }
diff --git a/RL2-Source/RLAudio/AmbientSoundController.cs b/RL2-Source-patched/RLAudio/AmbientSoundController.cs
index 71b357b..0181a2c 100755
--- a/RL2-Source/RLAudio/AmbientSoundController.cs
+++ b/RL2-Source-patched/RLAudio/AmbientSoundController.cs
@@ -57,7 +57,10 @@ public class AmbientSoundController : MonoBehaviour, IAudioEventEmitter
 
 		public override int GetHashCode()
 		{
-			return ((-1554893279 * -1521134295 + Biome.GetHashCode()) * -1521134295 + RoomType.GetHashCode()) * -1521134295 + IsLarge.GetHashCode();
+			unchecked
+			{
+				return ((-1554893279 * -1521134295 + Biome.GetHashCode()) * -1521134295 + RoomType.GetHashCode()) * -1521134295 + IsLarge.GetHashCode();
+			}
 		}
 	}
 
