diff --git a/RL2-Source/CharacterCreator.cs b/Assembly-CSharp/CharacterCreator.cs
index 50b3bfc..1a0f115 100644
--- a/RL2-Source/CharacterCreator.cs
+++ b/Assembly-CSharp/CharacterCreator.cs
@@ -1,3 +1,4 @@
+using RL2.ModLoader;
 using System;
 using System.Collections;
 using System.Collections.Generic;
@@ -135,6 +136,9 @@ public static class CharacterCreator
 			Debug.LogWarning("<color=yellow>Could not find non-duplicate talent in CharacterCreator.ApplyRandomizeKitTrait.</color>");
 		}
 		charData.Talent = abilityType3;
+		foreach (ModSystem modSystem in GameManager.Instance.GetComponents<ModSystem>()) {
+			modSystem.ModifyCharacterRandomization(charData);
+		}
 	}
 
 	public static void GenerateClass(ClassType classType, CharacterData charDataToMod)
