diff --git a/RL2-Source/EnemySpawnController.cs b/Assembly-CSharp/EnemySpawnController.cs
index 10af8cf..f365de1 100644
--- a/RL2-Source/EnemySpawnController.cs
+++ b/Assembly-CSharp/EnemySpawnController.cs
@@ -1,4 +1,6 @@
+using RL2.ModLoader;
 using System;
+using System.Linq;
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using UnityEngine;
@@ -534,6 +536,16 @@ public class EnemySpawnController : MonoBehaviour, ISpawnController, IHasProject
 			ResetEnemyPosition(EnemyInstance);
 			EnemyInstance.OnEnemyDeathRelay.AddListener(m_onEnemyDeath);
 			EnemyInstance.OnReactivationTimedOutRelay.AddListener(m_onEnemyTimedOut);
+			foreach (Type globalEnemy in ModLoader.LoadedGlobalEnemies)
+			{
+				GlobalEnemy globalEnemyInstance = EnemyInstance.GameObject.AddComponent(globalEnemy) as GlobalEnemy;
+				if (!globalEnemyInstance.AppliesToEnemyType.Contains(Type) || !globalEnemyInstance.AppliesToEnemyRank.Contains(Rank))
+				{
+					globalEnemyInstance.enabled = false;
+					continue;
+				}
+				globalEnemyInstance.OnSpawn();
+			}
 		}
 		else
 		{
