diff --git a/RL2-Source/GameManager.cs b/Assembly-CSharp/GameManager.cs
index 2cdc5a3..91ab0c1 100644
--- a/RL2-Source/GameManager.cs
+++ b/Assembly-CSharp/GameManager.cs
@@ -1,5 +1,6 @@
 using System.Collections;
 using RL_Windows;
+using RL2.ModLoader;
 using Steamworks;
 using UnityEngine;
 
@@ -47,6 +48,7 @@ public class GameManager : MonoBehaviour
 		{
 			Instance = this;
 			m_pauseEventArgs = new GamePauseStateChangeEventArgs(isPaused: false);
+			gameObject.AddComponent<UnityHook>();
 		}
 		else if (Instance != this)
 		{
