diff --git a/RL2-Source/GameManager.cs b/Assembly-CSharp/GameManager.cs
index 2cdc5a3..2f20e5f 100644
--- a/RL2-Source/GameManager.cs
+++ b/Assembly-CSharp/GameManager.cs
@@ -1,5 +1,6 @@
 using System.Collections;
 using RL_Windows;
+using RL2.ModLoader;
 using Steamworks;
 using UnityEngine;
 
@@ -15,13 +16,13 @@ public class GameManager : MonoBehaviour
 
 	public static bool IsApplicationClosing { get; private set; }
 
-	private static GameManager Instance
+	public static GameManager Instance
 	{
 		get
 		{
 			return m_instance;
 		}
-		set
+		private set
 		{
 			m_instance = value;
 		}
@@ -47,6 +48,8 @@ public class GameManager : MonoBehaviour
 		{
 			Instance = this;
 			m_pauseEventArgs = new GamePauseStateChangeEventArgs(isPaused: false);
+			gameObject.AddComponent<UnityHook>();
+			ModLoader.Log(ModLoader.LoadedModsNames);
 		}
 		else if (Instance != this)
 		{
