diff --git a/RL2-Source/LineageWindowController.cs b/Assembly-CSharp/LineageWindowController.cs
index da3b9b6..66a1b76 100644
--- a/RL2-Source/LineageWindowController.cs
+++ b/Assembly-CSharp/LineageWindowController.cs
@@ -9,6 +9,7 @@ using SceneManagement_RL;
 using TMPro;
 using UnityEngine;
 using UnityEngine.SceneManagement;
+using RL2.ModLoader;
 
 public class LineageWindowController : WindowController, IAudioEventEmitter, ILocalizable
 {
@@ -289,8 +290,11 @@ public class LineageWindowController : WindowController, IAudioEventEmitter, ILo
 			{
 				m_characterDataArray[i].TraitOne = TraitType.TrueRogue;
 				m_characterDataArray[i].AntiqueOneOwned = RelicType.None;
-			}
-			if (i == m_characterDataArray.Length - 1)
+            }
+            foreach (ModSystem modSystem in GameManager.Instance.GetComponents<ModSystem>()) {
+                modSystem.ModifyGeneratedCharacter(m_characterDataArray[i]);
+            }
+            if (i == m_characterDataArray.Length - 1)
 			{
 				bool flag = false;
 				SoulShopObj soulShopObj = SaveManager.ModeSaveData.GetSoulShopObj(SoulShopType.ChooseYourClass);
@@ -318,7 +322,7 @@ public class LineageWindowController : WindowController, IAudioEventEmitter, ILo
 						m_characterDataArray[i].AntiqueTwoOwned = RelicType.None;
 					}
 				}
-				SoulShopObj soulShopObj2 = SaveManager.ModeSaveData.GetSoulShopObj(SoulShopType.ChooseYourSpell);
+                SoulShopObj soulShopObj2 = SaveManager.ModeSaveData.GetSoulShopObj(SoulShopType.ChooseYourSpell);
 				if (!soulShopObj2.IsNativeNull() && soulShopObj2.CurrentEquippedLevel > 0)
 				{
 					AbilityType soulShopSpellChosen = SaveManager.ModeSaveData.SoulShopSpellChosen;
